### PR TITLE
improved clean up logic and addressed sync issues for large topologies

### PR DESCRIPTION
**Pull request by kingshukdev**
This PR makes the GRPC wire clean up more robust.
It addresses few intermittent synchronizations issues for GRPC wire in a large topology
This PR also address the issue of pod skip list growth when pods are deleted.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ test:
 # Build local binaries
 local-build:
 	CGO_ENABLED=0 GOOS=linux go build -o meshnet github.com/networkop/meshnet-cni/plugin 
-	CGO_ENABLED=0 GOOS=linux go build -o meshnetd github.com/networkop/meshnet-cni/daemon
+	CGO_ENABLED=1 GOOS=linux go build -o meshnetd github.com/networkop/meshnet-cni/daemon
```
Comment (networkop) [inline Makefile:25]: is there a reason why you changed that?

File: daemon/meshnet/handler.go [hunk 2]
```diff
@@ -108,6 +108,10 @@ func (m *Meshnet) SetAlive(ctx context.Context, pod *mpb.Pod) (*mpb.BoolResponse
 	return &mpb.BoolResponse{Response: true}, nil
 }
 
+// When a pod does nto find its peer to establish a link then it marks the peer as skipped.
+// The pod stores the list of skipped peers in it's skip list.
+// For example - when a high priority pod do not find it's peer to create the link then
```
Comment (networkop) [inline daemon/meshnet/handler.go:113]: I don't think there's any priority for pods. It's only about who comes up first while its peer is down

File: daemon/meshnet/handler.go [hunk 3]
```diff
@@ -141,22 +145,53 @@ func (m *Meshnet) Skip(ctx context.Context, skip *mpb.SkipQuery) (*mpb.BoolRespo
 	return &mpb.BoolResponse{Response: true}, nil
 }
 
+// This clean up function is called when a pod is getting deleted and it get called once for every link the pod has.
+// Anytime a pod is destroyed, for a link, it inserts itself in the skip list of its' peers on the link.
+// It removes the peer from its won skip list for this meshnet link
```
Comment (networkop) [inline daemon/meshnet/handler.go:150]: /won/own/

File: daemon/meshnet/meshnet.go [hunk 4]
```diff
@@ -79,18 +82,39 @@ func New(cfg Config) (*Meshnet, error) {
...
 		lis:     lis,
-		s:       newServerWithLogging(cfg.GRPCOpts...),
+		s:       svr,
 	}
 	mpb.RegisterLocalServer(m.s, m)
 	mpb.RegisterRemoteServer(m.s, m)
 	mpb.RegisterWireProtocolServer(m.s, m)
 	reflection.Register(m.s)
+
+	// After server is registered, reduce logging if link type is GRPC
```
Comment (networkop) [inline daemon/meshnet/meshnet.go:109]: what's the reason for this? would it make sense to just adhere to a global configuration option?

**Review by networkop — APPROVED** [review]
thanks @kingshukdev , most of it looks good and makes sense. I've just left a few comments where things weren't clear enough for me, would appreciate your feedback.

File: daemon/meshnet/handler.go (hunk 2 above)
Comment (kingshukdev) [reply daemon/meshnet/handler.go:113]: Correct. I should remove priority from that comment.

File: daemon/meshnet/meshnet.go (hunk 4 above)
Comment (kingshukdev) [reply daemon/meshnet/meshnet.go:109]: When we use VxLan then GRPC is meshnet control plane and with the current option for every GRPC RCP execution it log for SUCESS, WRANING and ERROR. This is reasonable amount of log as there are no too many RPC call. 

With GRPC wire GRPCP also be come the data plane. Then for every packet exchanged between  pods, GRPC generates a SUCCESS log for each packet transfer.  Per packet logs are too much logging and the rolling logs over write the previous logs very quickly. As a results we loose helpful information that is needed for debugging. So for GRPC data plane we have stopped all logs for now. In the next submit we intend to enable only ERROR log. 

For vxLAN it remains as is.

File: daemon/meshnet/handler.go (hunk 3 above)
Comment (kingshukdev) [reply daemon/meshnet/handler.go]: done.

**Comment by networkop** [comment]
great, thanks! will merge once the tests are passed

**Pull request merged**